core/src/main/scala/com/raphtory/api/analysis/table/TableOutputTracker.scala

```diff
@@ -98,26 +100,34 @@ case class TableOutputTracker(tracker: QueryProgressTracker, topics: TopicReposi
   def isJobDone: Boolean =
     outputDone
 
-  private def waitForNextResult(): Any =
+  private def waitForNextResult(): Any = {
+    logger.debug("waiting for result")
     if (timeout.isFinite)
       Option(completedResults.poll(timeout.length, timeout.unit))
     else
       Option(completedResults.take())
+  }
 
   override def hasNext: Boolean =
-    if (nextResult.isDefined)
+    if (nextResult.isDefined) {
+      logger.debug("hasNext true as result already present")
       true
-    else if (outputDone)
+    }
+    else if (outputDone) {
+      logger.debug("hasNext false as output complete")
       false
+    }
     else
       waitForNextResult() match {
         case Some(res) =>
           res match {
             case EndOutput      =>
               outputDone = true
+              logger.debug("hasNext false as output complete after waiting for result")
```

Comment:
are the debug statements really required?

Reply:
No debug statements are ever required...

As we currently are having some hanging issues when things go wrong, these are quite useful for knowing where we are...